library/src/engine.cxx

```diff
@@ -247,6 +250,46 @@ void engine::autoloadPlugins()
   factory::instance()->autoload();
 }
 
+//----------------------------------------------------------------------------
+std::vector<std::string> engine::getPluginsList(const std::string& pluginPath)
+{
+  vtksys::Directory dir;
+  constexpr std::string_view ext = ".json";
+  std::vector<std::string> pluginNames;
+
+  if (dir.Load(pluginPath))
+  {
+    for (unsigned long i = 0; i < dir.GetNumberOfFiles(); i++)
+    {
+      std::string currentFile = dir.GetFile(i);
+      if (std::equal(ext.rbegin(), ext.rend(), currentFile.rbegin()))
+      {
+        std::string fullPath = dir.GetPath();
+        fullPath += "/";
+        fullPath += currentFile;
```

Comment:
use filesystem

Reply:
There is no usage of filesystem in libf3d yet. We will do the migration later.

Reply:
indeed